agent/05_agent_configure.sh

```diff
@@ -245,7 +245,104 @@ function generate_cluster_manifests() {
   ansible-playbook -vvv \
           -e install_path=${SCRIPTDIR}/${INSTALL_CONFIG_PATH} \
           "${SCRIPTDIR}/agent/create-manifests-playbook.yaml"
+}
+
+function enable_load_balancer() {
+  if [[ "${PLATFORM_TYPE}" == "none" && "${NUM_MASTERS}" == "3" ]]; then
+    # setup haproxy as the load balancer
+    if [[ "${IP_STACK}" == "v6" ]]; then
+      # The "wildcard" is different depending on IP stack.
+      # See http://docs.haproxy.org/1.6/configuration.html#4.2-bind
+      export HAPROXY_WILDCARD="[::]"
+    else
+      export HAPROXY_WILDCARD="*"
+    fi
+
+    sudo dnf install -y haproxy
+    sudo rm -f /etc/haproxy/haproxy.cfg
+    sudo bash -c "cat << EOF >> /etc/haproxy/haproxy.cfg
+global
+  log         127.0.0.1 local2
+  pidfile     /var/run/haproxy.pid
+  maxconn     4000
+  daemon
+defaults
+  mode                    http
+  log                     global
+  option                  dontlognull
+  option http-server-close
+  option                  redispatch
+  retries                 3
+  timeout http-request    10s
+  timeout queue           1m
+  timeout connect         10s
+  timeout client          1m
+  timeout server          1m
+  timeout http-keep-alive 10s
+  timeout check           10s
+  maxconn                 3000
+frontend stats
+  bind *:1936
+  mode            http
+  log             global
+  maxconn 10
+  stats enable
+  stats hide-version
+  stats refresh 30s
+  stats show-node
+  stats show-desc Stats for ocp4 cluster
+  stats auth admin:ocp4
+  stats uri /stats
+listen api-server-6443
+  bind ${HAPROXY_WILDCARD}:6443
+  mode tcp
+  server master-0 master-0.${CLUSTER_DOMAIN}:6443 check inter 1s
+  server master-1 master-1.${CLUSTER_DOMAIN}:6443 check inter 1s
+  server master-2 master-2.${CLUSTER_DOMAIN}:6443 check inter 1s
```

Comment:
It feels like an iteration would be required here (and below), based on NUM_MASTERS and NUM_WORKERS